code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -338,84 +328,79 @@
 				if(2)
 					var/scaled_production = clamp(heat_output * 1e-3, 0, fuel_consumption_rate) * delta_time
-					moderator_internal.gases[/datum/gas/carbon_dioxide][MOLES] += scaled_production * 1.65
-					moderator_internal.gases[/datum/gas/water_vapor][MOLES] += scaled_production
-					if(m_plasma > 50)
+					moderator_internal.gases[selected_fuel.secondary_products[1]][MOLES] += scaled_production * 1.65
+					moderator_internal.gases[selected_fuel.secondary_products[2]][MOLES] += scaled_production
+					if(moderator_list[/datum/gas/plasma] > 50)
 						internal_output.assert_gases(/datum/gas/bz)
 						internal_output.gases[/datum/gas/bz][MOLES] += scaled_production * 1.8
-						moderator_internal.gases[/datum/gas/nitrous_oxide] += scaled_production * 1.15
+						moderator_internal.gases[selected_fuel.secondary_products[3]] += scaled_production * 1.15
 						moderator_internal.gases[/datum/gas/plasma][MOLES] -= min(moderator_internal.gases[/datum/gas/plasma][MOLES], scaled_production * 1.75)
-					if(m_proto_nitrate > 20)
+					if(moderator_list[/datum/gas/proto_nitrate] > 20)
 						radiation *= 1.55
 						heat_output *= 1.025
 						internal_output.assert_gases(/datum/gas/stimulum)
 						internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 1.05
-						moderator_internal.gases[/datum/gas/plasma][MOLES] += scaled_production * 1.65
 						moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 1.35)
-					if(m_n2o > 50)
-						radiation *= 0.55
-						heat_output *= 1.055
-						moderator_internal.gases[/datum/gas/halon] += scaled_production * 1.35
-						moderator_internal.gases[/datum/gas/nitrous_oxide][MOLES] -= min(moderator_internal.gases[/datum/gas/nitrous_oxide][MOLES], scaled_production * 1.5)
 				if(3, 4)
 					var/scaled_production = clamp(heat_output * 5e-4, 0, fuel_consumption_rate) * delta_time
 					if(power_level == 3)
-						moderator_internal.gases[/datum/gas/oxygen][MOLES] += scaled_production * 0.5
-						moderator_internal.gases[/datum/gas/nitrogen][MOLES] += scaled_production * 0.45
+						moderator_internal.gases[selected_fuel.secondary_products[2]][MOLES] += scaled_production * 0.5
+						moderator_internal.gases[selected_fuel.secondary_products[3]][MOLES] += scaled_production * 0.45
 					if(power_level == 4)
-						moderator_internal.gases[/datum/gas/carbon_dioxide][MOLES] += scaled_production * 1.65
-						moderator_internal.gases[/datum/gas/water_vapor][MOLES] += scaled_production * 1.25
-					if(m_plasma > 10)
-						moderator_internal.gases[/datum/gas/bz][MOLES] += scaled_production * 1.1
+						moderator_internal.gases[selected_fuel.secondary_products[3]][MOLES] += scaled_production * 1.65
+						moderator_internal.gases[selected_fuel.secondary_products[4]][MOLES] += scaled_production * 1.25
```

Comment:
and the two/four?

Reply:
what about 2/4? those are on different power levels, 2 is on power level 2 and 3, 4 is on power level 4 and 5